quicktree.php

```diff
@@ -4,226 +4,406 @@
+			</td>
+		</tr></p>';
+
+		print '</table></td></tr>';
+
+		print '<tr><td class="saveRow">
+			<input type="hidden" name="action" value="save">
+			<input type="button" value="' . __esc('Cancel') . '" onClick="cactiReturnTo()">&nbsp;<input type="submit" value="' . __esc('Continue') . '" title="' . __esc('Add To Branch', 'quicktree') . '">
+		</td></tr>';
+
+		html_end_box();
+
+		form_end();
+
+		break;
+	case 'add_branch':
+		if (get_nfilter_request_var('header') == null) {
+			top_header();
+		}
+
+		form_start('quicktree.php', 'quicktree_form');
+
+		html_start_box($form_actions[$drp_action], '60%', '', '3', 'center', '');
+
+		$queryrows = db_fetch_assoc("SELECT g.id, g.name
+			FROM graph_tree AS g
+			ORDER BY g.name");
+
+		print '<tr><td>';
+
+		print '<p>' . __('Click \'Continue\' to Add the following Tree Branch.', 'quicktree') . '</p>';
+
+		print '<p>' . __('Add to which Graph Tree?', 'quicktree') . '</p>';
+
+		print '<table class="filterTable">';
+
+		print '<tr>
+			<td>
+				<p><label for="tree_id">' . __('Tree Name', 'quicktree') . '</label></p>
+			</td>
+			<td>
+				<select id="tree_id" name="tree_id">';
+
+				if (cacti_sizeof($queryrows)) {
+					foreach ($queryrows as $tr) {
+						printf('<option value="%d">%s</option>', $tr['id'], html_escape($tr['name']));
+					}
+				}
+
+				print '</select>
+			</td>
+		</tr>';
+
+		print '<tr>
+			<td>
+				<p><label for="branch">' .__('Branch Name', 'quicktree') . '</label></p>
+			</td>
+			<td>
+				<input class="ui-state-default ui-corner-all" id="branch" name="branch" value="' . __('QuickTree', 'quicktree') . '">
+			</td>
+		</tr></p>';
+
+		print '</table></td></tr>';
+
+		print '<tr><td class="saveRow">
+			<input type="hidden" name="action" value="save">
+			<input type="button" value="' . __esc('Cancel') . '" onClick="cactiReturnTo()"><input type="submit" value="' . __esc('Continue') . '" title="' . __esc('Add To Branch', 'quicktree') . '">
+		</td></tr>';
+
+		html_end_box();
+
+		form_end();
+
+		break;
+	case 'clear':
+		$SQL = db_execute_prepared('DELETE FROM quicktree_graphs WHERE userid = ?', array($user));
+
+		header('Location: quicktree.php?header=false&drp_action=&action=&location=' . get_nfilter_request_var('location'));
+
+		break;
+	case 'save':
+		$new_tree_id = -1;
+		$parent_id   = 0;
+
+		$username = db_fetch_cell_prepared('SELECT username
+			FROM user_auth
+			WHERE id = ?',
+			array($user));
+
+		if (isset_request_var('tree_id')) {
+			$new_tree_id = get_filter_request_var('tree_id');
+
+			$tree_name = db_fetch_cell_prepared('SELECT name
+				FROM graph_tree
+				WHERE id = ?',
+				array($new_tree_id));
+		} elseif (isset_request_var('tree')) {
+			$new_tree_id = db_fetch_cell_prepared('SELECT id
+				FROM graph_tree
+				WHERE name = ?',
+				array(get_nfilter_request_var('tree')));
+
+			$tree_name = get_nfilter_request_var('tree');
+		}
+
+		if ($tree_name == '') {
```

Comment:
This should really be `if (empty($tree_name))` to avoid any spaces being simply treated as a name.